Navitia.php

```diff
@@ -2,93 +2,78 @@
-use Canaltp\NavitiaPhp\Exception\NavitiaException;
+use CanalTP\NavitiaPhp\Exception\NavitiaException;
 
 class Navitia
 {
+    /**
+     * @var Client
+     */
     private $guzzle;
-    private $defaultCoverage;
 
     /**
      * Navitia constructor.
+     *
+     * @param Client $guzzle
      */
-    public function __construct(Client $guzzle, $coverage)
+    public function __construct(Client $guzzle)
     {
         $this->guzzle = $guzzle;
-        $this->defaultCoverage = $coverage;
     }
 
     /**
-     * @return mixed
+     * @param string $navitiaBaseUrl
+     * @param string $token
+     *
+     * @return self
      */
-    public function getDefaultCoverage()
+    public static function createFromBaseUrlAndToken($navitiaBaseUrl, $token)
     {
-        return $this->defaultCoverage;
+        $guzzle = new Client([
+            'base_uri' => $navitiaBaseUrl,
+            'auth' => [$token, ''],
+        ]);
+
+        return new self($guzzle);
     }
 
-    public function call($options)
+    /**
+     * @return Client
+     */
+    public function getGuzzleClient()
     {
-        try
-        {
-            $this->configureOptions($options);
-
-            $result = $this->guzzle->get($this->buildUrl($options));
-            if (200 !== $result->getStatusCode()) {
-
-                throw new NavitiaException();
-            }
-            
-            return Yaml::parse((string) $result->getBody());
-        }
-        catch (ClientException $e) {
-            dump('verify your request : 404');
-            dump($e);
-        }
-        catch (ServerException $e) {
-            dump('Navitia is down : 500');
-            dump($e);
-        }
-        catch (RequestException $e) {
-            dump('Something wrong happen');
-            dump($e);
-        }
+        return $this->guzzle;
     }
 
-    public function configureOptions(&$options)
+    /**
+     * @param Client $guzzle
+     *
+     * @return self
+     */
+    public function setGuzzleClient(Client $guzzle)
     {
+        $this->guzzle = $guzzle;
 
-        $resolver = new OptionsResolver();
-        $resolver->setDefaults(array(
-            'coverage' => $this->defaultCoverage,
-            'path'     => null,
-            'parameters' => null
-        ));
-
-        $options = $resolver->resolve($options);
+        return $this;
     }
 
-    private function buildUrl($options)
+    /**
+     * @param sring $path
+     *
+     * @return \stdClass
+     *
+     * @throws NavitiaException
+     */
+    public function call($path)
     {
-        $url = 'coverage/'. $options['coverage'] .'/';
+        try {
+            $result = $this->guzzle->get($path);
 
-        if (!empty($options['coverage'])) {
-            foreach ($options['path'] as $item) {
-                $url += $item['element'] .'/'. $item['value']. '/';
-            }
+            return json_decode($result->getBody());
+        } catch (ClientException $e) {
```

Comment:
Why do you catch all guzzle exception, it reduce Guzzle explanation, be more explicit or let guzzle exception go through you class